File: PlanIT-backend/src/main/java/com/example/demo/model/Message.java
package com.example.demo.model;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;

@Entity
@Table(name = "messages")
public class Message {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;

    String message;

    @ManyToOne
    @JsonIgnore
    @JoinColumn(name="room_id")
    private Room room;

    private String username;

    public Message() {
    }

    public Message(String message, Room room, String username) {
        this.message = message;
        this.room = room;
        this.username = username;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public Room getRoom() {
        return room;
    }

    public void setRoom(Room room) {
        this.room = room;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    @Override
    public String toString() {
        return "Message{" +
                "id=" + id +
                ", message='" + message + '\'' +
                ", room=" + room +
                ", username='" + username + '\'' +
                '}';
    }
}

File: PlanIT-frontend/src/app/model/Message.ts
export class Message{

  private _message: string;
  private _username: string;


  constructor(message: string, username: string) {
    this._message = message;
    this._username = username;
  }


  get message(): string {
    return this._message;
  }

  set message(value: string) {
    this._message = value;
  }

  get username(): string {
    return this._username;
  }

  set username(value: string) {
    this._username = value;
  }
}
